content/en/docs/concepts/storage/volumes.md

```diff
@@ -11,221 +11,289 @@ weight: 10
+spec:
+    containers:
+    - name: mysql
+      image: mysql
+      env:
+      - name: MYSQL_ROOT_PASSWORD
+        value: "rootpasswd"
+      volumeMounts:
+      - mountPath: /var/lib/mysql
+        name: site-data
+        subPath: mysql
+    - name: php
+      image: php:7.0-apache
+      volumeMounts:
+      - mountPath: /var/www/html
+        name: site-data
+        subPath: html
+    volumes:
+    - name: site-data
+      persistentVolumeClaim:
+        claimName: my-lamp-site-data
+```
+
+### Using subPath with expanded environment variables {#using-subpath-expanded-environment}
+
+{{< feature-state for_k8s_version="v1.17" state="stable" >}}
+
+Use the `subPathExpr` field to construct `subPath` directory names from
+downward API environment variables. The `subPath` and `subPathExpr` properties are mutually exclusive.
+
+In this example, a `Pod` uses `subPathExpr` to create a directory `pod1` within
+the `hostPath` volume `/var/log/pods`.
+The `hostPath` volume takes the `Pod` name from the `downwardAPI`.
+The host directory `/var/log/pods/pod1` is mounted at `/logs` in the container.
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: pod1
 spec:
   containers:
-  - image: k8s.gcr.io/test-webserver
-    name: test-cinder-container
+  - name: container1
+    env:
+    - name: POD_NAME
+      valueFrom:
+        fieldRef:
+          apiVersion: v1
+          fieldPath: metadata.name
+    image: busybox
+    command: [ "sh", "-c", "while [ true ]; do echo 'Hello'; sleep 10; done | tee -a /logs/hello.txt" ]
     volumeMounts:
-    - mountPath: /test-cinder
-      name: test-volume
+    - name: workdir1
+      mountPath: /logs
+      subPathExpr: $(POD_NAME)
+  restartPolicy: Never
   volumes:
-  - name: test-volume
-    # This OpenStack volume must already exist.
-    cinder:
-      volumeID: "<volume id>"
-      fsType: ext4
+  - name: workdir1
+    hostPath:
+      path: /var/log/pods
 ```
 
-#### OpenStack CSI migration
+## Core Volume Types
 
-{{< feature-state for_k8s_version="v1.18" state="beta" >}}
+### cephfs
 
-The `CSIMigration` feature for Cinder, when enabled, redirects all plugin operations
-from the existing in-tree plugin to the `cinder.csi.openstack.org` Container
-Storage Interface (CSI) Driver. In order to use this feature, the [Openstack Cinder CSI
-Driver](https://github.com/kubernetes/cloud-provider-openstack/blob/master/docs/using-cinder-csi-plugin.md)
-must be installed on the cluster and the `CSIMigration` and `CSIMigrationOpenStack`
-beta features must be enabled.
+A `cephfs` volume allows an existing CephFS volume to be  mounted into your pod. CephFS can be mounted by multiple writers simultaneously.
+
+See the [CephFS example](https://github.com/kubernetes/examples/tree/{{< param "githubbranch" >}}/volumes/cephfs/) for more details.
 
 ### configMap
 
-A [ConfigMap](/docs/tasks/configure-pod-container/configure-pod-configmap/)
+A [ConfigMap](/docs/tasks/configure-pod-container/configure-pod-configmap/) 
```

Comment:
BTW this could link to https://kubernetes.io/docs/concepts/configuration/configmap/